java/arcs/core/data/proto/policy.proto

```diff
@@ -0,0 +1,107 @@
+syntax = "proto3";
+
+package arcs;
+
+option java_package = "arcs.core.data.proto";
+option java_multiple_files = true;
+
+// Defines a data usage policy.
+message PolicyProto {
+  // Policy name.
+  string name = 1;
+
+  // Targets governed by the policy.
+  repeated PolicyTargetProto targets = 2;
+
+  // Additional config metadata for the policy.
+  repeated PolicyConfigProto configs = 3;
+
+  // Human readable description for the policy.
+  string description = 4;
+
+  enum EgressType {
+    EGRESS_TYPE_UNSPECIFIED = 0;
+    LOGGING = 1;
+    FEDERATED_ANALYTICS = 2;
+  }
+  // Egress type permitted by the policy.
+  EgressType egress_type = 5;
+
+  // Whether user consent is required before using the policy.
+  bool user_consent_required = 6;
```

Comment:
Anything which is going to affect the way the policy translator works should be called out in the proto explicitly. Otherwise operating the translator will require a bunch of necessary data that is hidden away in the proto inside untyped and undocumented metadata fields. It's much better to call them out explicitly.

The only things which should go in the client config/metadata fields are things that are only meant to be plumbed to the client and which don't affect the operation of the policy engine.

Reply:
But why does the translator need to know about this? Isn't this something very specific to the client? Won't you need to know details about the client to "enforce" this (e.g. where is the user consent "stored"? In fact while a bool is good enough for our immediate use case, it doesn't explain what type of user consent is required and we are already planning to support different types of user consent in the not-too-distant future

Reply:
I'll remove this for now then. I had thought this would required dedicated support in the policy translator, but I guess we can wait and see.